File: src/main/java/io/colyseus/example/java/MyState.java
package io.colyseus.example.java;

import io.colyseus.annotations.SchemaField;
import io.colyseus.serializer.schema.Schema;
import io.colyseus.serializer.schema.types.ArraySchema;
import io.colyseus.serializer.schema.types.MapSchema;

public class MyState extends Schema {
    @SchemaField(v1 = "0/ref", v2 = PrimitivesTest.class)
    public PrimitivesTest primitives = new PrimitivesTest();

    @SchemaField(v1 = "1/array/ref", v2 = Player.class)
    public ArraySchema<Player> players = new ArraySchema<>(Player.class);

    @SchemaField(v1 = "2/map/ref", v2 = Cell.class)
    public MapSchema<Cell> cells = new MapSchema<>(Cell.class);
}

class Cell extends Schema {
    @SchemaField(v1 = "0/float32")
    public float x;

    @SchemaField(v1 = "1/float32")
    public float y;
}

class PrimitivesTest extends Schema {
    @SchemaField(v1 = "0/uint8")
    public short _uint8;

    @SchemaField(v1 = "1/uint16")
    public int _uint16;

    @SchemaField(v1 = "2/uint32")
    public long _uint32;

    @SchemaField(v1 = "3/uint64")
    public long _uint64;

    @SchemaField(v1 = "4/int8")
    public byte _int8;

    @SchemaField(v1 = "5/int16")
    public short _int16;

    @SchemaField(v1 = "6/int32")
    public int _int32;

    @SchemaField(v1 = "7/int64")
    public long _int64;

    @SchemaField(v1 = "8/float32")
    public float _float32_n;

    @SchemaField(v1 = "9/float32")
    public float _float32_p;

    @SchemaField(v1 = "10/float64")
    public double _float64_n;

    @SchemaField(v1 = "11/float64")
    public double _float64_p;

    @SchemaField(v1 = "12/boolean")
    public boolean _boolean;

    @SchemaField(v1 = "13/string")
    public String _string;
}

class Player extends Schema {
    @SchemaField(v1 = "0/int32")
    public int x;
}

File: src/main/java/io/colyseus/example/kotlin/MyState.kt
package io.colyseus.example.kotlin

import io.colyseus.serializer.schema.Schema
import io.colyseus.annotations.SchemaField
import io.colyseus.default
import io.colyseus.serializer.schema.types.ArraySchema
import io.colyseus.serializer.schema.types.MapSchema

class MyState : Schema() {
    @SchemaField("0/ref", PrimitivesTest::class)
    var primitives = PrimitivesTest()

    @SchemaField("1/array/ref", Player::class)
    var players = ArraySchema(Player::class.java)

    @SchemaField("2/map/ref", Cell::class)
    var cells = MapSchema(Cell::class.java)
}

class Cell : Schema() {
    @SchemaField("0/float32")
    var x = Float.default

    @SchemaField("1/float32")
    var y = Float.default
}

class PrimitivesTest : Schema() {
    @SchemaField("0/uint8")
    var _uint8 = Short.default

    @SchemaField("1/uint16")
    var _uint16 = Int.default

    @SchemaField("2/uint32")
    var _uint32 = Long.default

    @SchemaField("3/uint64")
    var _uint64 = Long.default

    @SchemaField("4/int8")
    var _int8 = Byte.default

    @SchemaField("5/int16")
    var _int16 = Short.default

    @SchemaField("6/int32")
    var _int32 = Int.default

    @SchemaField("7/int64")
    var _int64 = Long.default

    @SchemaField("8/float32")
    var _float32_n = Float.default

    @SchemaField("9/float32")
    var _float32_p = Float.default

    @SchemaField("10/float64")
    var _float64_n = Double.default

    @SchemaField("11/float64")
    var _float64_p = Double.default

    @SchemaField("12/boolean")
    var _boolean = Boolean.default

    @SchemaField("13/string")
    var _string = String.default
}

class Player : Schema() {
    @SchemaField("0/int32")
    var x = Int.default
}